### PR TITLE
DataFrame.join()

### PR DESCRIPTION
**Pull request by yuwillrun**

### CODE REVIEW

**Comment by yuwillrun** [comment]
@ueshin need to code review.Thanks~~

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -3222,6 +3222,113 @@ def merge(left, right: 'DataFrame', how: str = 'inner',
         else:
             return DataFrame(selected_columns)
 
+    def join(left, right: 'DataFrame', on: Optional[Union[str, List[str]]] = None,
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: Seems like a mistake. left -> self. I think it's a Python convention.

File: databricks/koalas/frame.py [hunk 2]
```diff
@@ -3222,6 +3222,113 @@ def merge(left, right: 'DataFrame', how: str = 'inner',
...
+        DataFrame.merge: For column(s)-on-columns(s) operations.
+
+        Notes
+        -----
+        Parameters on, lsuffix, and rsuffix are not supported when passing a list of DataFrame
+        objects.
+
+        Examples
+        --------
+        >>> kdf1 = ks.DataFrame({'key': ['K0', 'K1', 'K2', 'K3'],
```
Comment (HyukjinKwon) [inline databricks/koalas/frame.py]: Out of curiosity, why did we use a dataset different from Pandas? (https://github.com/pandas-dev/pandas/blob/v0.24.2/pandas/core/frame.py#L6694-L6815)

**Comment by HyukjinKwon** [comment]
Looks good to me otherwise

File: databricks/koalas/frame.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/frame.py]: oh, I used `left` and `right` when I updated `merge` function before. I'll submit the PR to fix `merge` function soon.

File: databricks/koalas/frame.py (hunk 2 above)
Comment (yuwillrun) [reply databricks/koalas/frame.py]: In another PR, i use pandas example, but be told the example is too complex during code review.So in this PR, i simplify the redundant rows.Which way do you think is better，same pandas or simplify examlple ?

File: databricks/koalas/frame.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/frame.py]: FYI: the PR #479 was merged.

**Comment by yuwillrun** [comment]
> Looks good to me otherwise

Fix already.

**Comment by yuwillrun** [comment]
@ueshin Need to merge.~~

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**